src/Garden/SearchView/PlantSearchView.jsx

```diff
@@ -1,98 +1,84 @@
+import PaginationComponent from "../../HelperComponents/PaginationComponent";
 import PlantSearchClient from "../clients/plantSearchClient";
+
 const plantSearchClient = new PlantSearchClient();
 
-class PlantSearchView extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            searchTerm: '',
-            searchPlants: [],
-            totalPlants: 0,
-            itemsPerPage: 0,
-            lastPage: 0,
-            paginationSearchError: null
-        }
-    }
-    
-    setSearchPlants = (searchTerm, response) => {
-        response && this.setState({
-            searchTerm, 
-            searchPlants: response.data, 
-            totalPlants: response.total, 
-            itemsPerPage: response.perPage,
-            lastPage: response.lastPage,
-            paginationSearchError: null
-        });
+const PlantSearchView = (props) => {
+    const{selectedGarden, selectedGardenPlants, addPlantToGarden} = props;
+    const [searchTerm, setSearchTerm] = useState('');
+    const [searchPlants, setSearchPlants] = useState([]);
+    const [totalPlants, setTotalPlants] = useState(0);
+    const [itemsPerPage, setItemsPerPage] = useState(0);
+    const [lastPage, setLastPage] = useState(0);
+    const [paginationSearchError, setPaginationSearchError] = useState(null);
+
+    const searchPlantResults = (searchTerm, response) => {
+        if (response){
+            setSearchTerm(searchTerm);
+            setSearchPlants(response.data);
+            setTotalPlants(response.total); 
+            setItemsPerPage(response.perPage);
+            setLastPage(response.lastPage);
+            setPaginationSearchError(null);
+        };       
     }
 
-    paginateSearchPlants = (page) => {
+    const paginateSearchPlants = (page) => {
         console.log('paginateSearchPlants', page);
-        plantSearchClient.search(this.state.searchTerm, page)
-        .then(response => this.handlePaginationResponse(response))
+        plantSearchClient.search(searchTerm, page)
+        .then(response => handlePaginationResponse(response))
         .catch(error => { 
             console.log(error); 
             // Retry once
-            plantSearchClient.search(this.state.searchTerm, page)
-                .then(response => this.handlePaginationResponse(response))
+            plantSearchClient.search(searchTerm, page)
+                .then(response => handlePaginationResponse(response))
                 .catch(error => {
                     console.log(error);
-                    this.setState({paginationSearchError: 'There was an error searching for plants. Please try again.'});
+                    setPaginationSearchError('There was an error searching for plants. Please try again.');
                 });
         });
-    }
+    };
+    //helper function
+    const handlePaginationResponse = (response) => {  
+        console.log('paginateSearchPlants', response);
+        if (response !== null){
```

Comment:
```suggestion
        if (response){
```

Remember this was 'undefined' instead of null? both null and undefined will return false, so we can switch to just look at the response to capture both situations.